homeassistant/components/media_player/denon.py

```diff
@@ -122,17 +152,28 @@ def is_volume_muted(self):
     @property
     def source_list(self):
         """List of available input sources."""
-        return list(self._source_list.keys())
+        # TODO: Add filtering/rename posibility?
+        return sorted(list(self._source_list.keys()))
 
     @property
     def media_title(self):
-        """Current media source."""
-        return self._mediasource
+        """Current media info."""
+        return self._mediainfo
 
     @property
     def supported_media_commands(self):
         """Flag of media commands that are supported."""
-        return SUPPORT_DENON
+        if self._mediasource in MEDIA_MODES.values():
+            return SUPPORT_DENON | SUPPORT_MEDIA_MODES
+        else:
+            return SUPPORT_DENON
+
+    @property
+    def source(self):
+        """Return the current input source."""
+        for pretty_name, name in self._source_list.items():
```

Comment:
In my opinion, the python syntax for generator expressions and list or dict comprehensions is so good that you can understand them at first encounter. It's one elegant line that is not more convoluted than what would be written in three lines.